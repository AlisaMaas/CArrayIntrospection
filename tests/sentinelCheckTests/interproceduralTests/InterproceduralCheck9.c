/**
 * This tests that the case in which the second
 * of several calls determines that the argument
 * is a null-terminated array. foo and find's formal
 * parameters "string" should be null-terminated.
 **/
void print(char *string) {}
int find(char nonString[], char string[]) {
	print(nonString);
	for (int i = 0; i < 10; i++) {
		if (i == 4) {
			if (string[i] == '\0') {
				break;
			}
		}
	}
	return 1;
}
int foo(char nonString[], char string[]) {
	return find("hello", string);
}
