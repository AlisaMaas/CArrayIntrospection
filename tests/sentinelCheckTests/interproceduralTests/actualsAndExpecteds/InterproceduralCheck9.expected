Printing analysis 'Promote Memory to Register' for function 'print':
Pass::print not implemented for pass: 'Promote Memory to Register'!
Printing analysis 'Promote Memory to Register' for function 'find':
Pass::print not implemented for pass: 'Promote Memory to Register'!
Printing analysis 'Promote Memory to Register' for function 'foo':
Pass::print not implemented for pass: 'Promote Memory to Register'!
Printing analysis 'Determine whether and how to annotate each function with the null-terminated annotation':
find with argument 1 should not be annotated NULL_TERMINATED (1).
foo with argument 1 should not be annotated NULL_TERMINATED (0).
